plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsSecurityContext.java

```
@@ -0,0 +1,153 @@
+            new ReflectPermission("suppressAccessChecks"),
+            // 2) allow hadoop to add credentials to our Subject
+            new AuthPermission("modifyPrivateCredentials"),
+            // 3) allow hadoop to act as the logged in Subject
+            new AuthPermission("doAs"),
+            // 4) Listen and resolve permissions for kerberos server principals
+            new SocketPermission("localhost:0", "listen,resolve"),
+            // We add the following since hadoop requires the client to re-login when the kerberos ticket expires:
+            // 5) All the permissions needed for UGI to do it's weird JAAS hack
+            new RuntimePermission("getClassLoader"),
+            new RuntimePermission("setContextClassLoader"),
+            // 6) Additional permissions for the login modules
+            new AuthPermission("modifyPrincipals"),
+            new PrivateCredentialPermission("org.apache.hadoop.security.Credentials * \"*\"", "read"),
+            new PrivateCredentialPermission("javax.security.auth.kerberos.KerberosTicket * \"*\"", "read"),
+            new PrivateCredentialPermission("javax.security.auth.kerberos.KeyTab * \"*\"", "read")
+            // Included later:
+            // 7) allow code to initiate kerberos connections as the logged in user
+            // Still far and away fewer permissions than the original full plugin policy
+        };
+    }
+
+    /**
+     * Locates the keytab file in the environment and verifies that it exists.
+     * Expects keytab file to exist at {@code $CONFIG_DIR$/kerberos/krb5.keytab}
+     */
+    public static Path locateKeytabFile(Environment environment) {
+        Path keytabPath = environment.configFile().resolve("repository-hdfs").resolve("krb5.keytab");
+        try {
+            if (Files.exists(keytabPath) == false) {
+                LOGGER.error("Could not locate keytab file at [{}].", keytabPath);
+                throw new RuntimeException("Could not locate keytab");
```

Include the path in the exception message? I question whether the logging statement is needed given that the exception message will effectively say the same thing? Can you throw this as a FileNotFoundException from here and let it bubble all the way up to a higher layer for wrapping?